### PR TITLE
Add serializing/unserializing of "locked" actor attribute

### PR DESCRIPTION
**Pull request by Gargron**

### CODE REVIEW

**Comment by nightpool** [comment]
what's the rationale for using `_:locked`? the JSON-LD standard only ever uses `_:` for the creation of `@id`s in the presence of cyclic data, never as keys.

**Comment by Gargron** [comment]
@nightpool According to @puckipedia `_:locked` and `locked` are equivalent because it's not a registered property of any kind. For me personally it's better to keep track of which properties are not part of ActivityPub.

**Comment by nightpool** [comment]
mmm. okay. I think I can see how this works. I'm down for whatever, but it seems more explicit to me to use something like `https://joinmastodon.org/schema/locked` as your key to avoid conflicts/be more JSON-LD

File: app/lib/activitypub/case_transform.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      @camel_lower_cache ||= {}
+    end
+
+    def camel_lower(value)
+      case value
+      when Array then value.map { |item| camel_lower(item) }
+      when Hash then value.deep_transform_keys! { |key| camel_lower(key) }
+      when Symbol then camel_lower(value.to_s).to_sym
+      when String
+        camel_lower_cache[value] ||= if value.start_with?('_:')
```
Comment (nightpool) [inline app/lib/activitypub/case_transform.rb]: this should probably be general enough to handle other compact IRIs, like `foaf:name` (the first example from the JSON-LD spec, chosen at random)

**Review by nightpool — CHANGES_REQUESTED** [review]
I'm fine with punting on namespacing. LGTM w/ camel_lower generalized just a *tiny* bit.

File: app/lib/activitypub/case_transform.rb (hunk 1 above)
Comment (nightpool) [inline app/lib/activitypub/case_transform.rb]: maybe something like `.split(':').map {|i| i.underscore.camelize(:lower)}.join(':')`?

**Review by nightpool — APPROVED** [review]
Nevermind, looks like underscore.camelize(:lower) already does the right thing and I was confused.

**Pull request merged**